flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/InternalSourceSplitMetricGroup.java

```diff
@@ -70,9 +70,11 @@ private InternalSourceSplitMetricGroup(
                 MetricNames.ACC_SPLIT_ACTIVE_TIME, this::getAccumulatedActiveTime);
         splitWatermarkMetricGroup.gauge(
                 MetricNames.ACC_SPLIT_IDLE_TIME, this::getAccumulatedIdleTime);
-        currentWatermarkGauge =
-                splitWatermarkMetricGroup.gauge(
-                        MetricNames.SPLIT_CURRENT_WATERMARK, currentWatermark);
+        if (currentWatermark != null) {
+            currentWatermarkGauge =
+                    splitWatermarkMetricGroup.gauge(
+                            MetricNames.SPLIT_CURRENT_WATERMARK, currentWatermark);
+        }
```

Comment:
I think those changes are not needed right? If so, can you revert them + restore `final` for `currentWatermarkGauge`?